### ISSUE TITLE
certain commands fail to work with Vim 8.2.2350

### ISSUE THREAD

**Issue by fatih** [fatih/vim-go#3138]
### What did you do? (required: The issue will be **closed** when not provided)

I run `:GoDecls` and `:GoDef` on certain identifiers. They don't work on `8.2.2350`. But they work when I downgrade vim to `8.2.2300`. Something between `8.2.2300` and `8.2.2350` is causing `vim-go` to error for these commands. 

### What did you expect to happen?

I expect the commands to work.

### What happened instead?

vim-go throws an error. Here is the log from `:Messages`:

```
  1 || Messages maintainer: Bram Moolenaar <Bram@vim.org>
  2 || "main.go" 905L, 20989B
  3 || vim-go: initializing gopls
  4 || "main.go" 905L, 20989B
  5 || Error detected while processing function go#decls#Decls[10]..fzf#decls#cmd[12]..fzf#run[68]..function
  6 || go#decls#Decls[10]
  7 || fzf#decls#cmd[12]
  8 || fzf#run[64]
  9 || <SNR>48_callback[21]
 10 || Vim(call):E474: Invalid argument
 11 || Error detected while processing function
 12 || go#decls#Decls[10]
 13 || E171: Missing :endif
 14 || vim-go: initialized gopls
 15 || "main_test.go" 904L, 17479B
~
```

### Configuration (**MUST** fill this out):

 My vimrc config for vim-go starts here: https://github.com/fatih/dotfiles/blob/master/vimrc#L408

#### vim-go version:

I'm using `vim-go` with the least HEAD `3a8e3f6ded412b25fb3c13eecf76d5809fa175e8`.

####  `vimrc` you used to reproduce:

You can use my dotfiles here:  My vimrc config for vim-go starts here: https://github.com/fatih/dotfiles/blob/master/vimrc#L408

#### Vim version (first three lines from `:version`):

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan 14 2021 23:04:22)
macOS version - x86_64
Included patches: 1-2350
```
#### Go version (`go version`):

```
$ go version
go version go1.15.6 darwin/amd64
```

**Comment by bhcleek** [comment]
This was due to a problem introduced in Vim itself. It's been fixed in https://github.com/vim/vim/releases/tag/v8.2.2363.

I will leave this issue pinned until homebrew updates the [Vim recipe](https://formulae.brew.sh/formula/vim) to at least 8.2.2363.

**Comment by bhcleek** [comment]
I've created a PR to homebrew to try to get a new version before the usual version boundary that homebrew uses: https://github.com/Homebrew/homebrew-core/pull/69328.

In the meantime, brew users can update their local formula:
# edit $(brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/vim.rb
# update the formula to use at least 8.2.2363. Here's an example with 8.2.2375
```
url "https://github.com/vim/vim/archive/v8.2.2375.tar.gz"
sha256 "54d6c1aa46857c734f1a5f9f89414046266d678bd71ad94f7a961a2547ec0eda"
```
# uninstall Vim and reinstall from source:
```
brew uninstall --force vim
brew install --build-from-source vim
```

More information about the process to update a formula can be found in homebrew's [CONTRIBUTING.md](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md#to-contribute-a-fix-to-the-foo-formula)

**Comment by bhcleek** [comment]
The fix has been applied to  homebrew. The usual `brew upgrade vim` should suffice to to get the fix now.

I'll leave this open for a day or two so others have a chance to see it.

**Comment by gchaincl** [comment]
Confirmed! I had the same issue on ArchLinux, which got resolved after upgrading vim to `8.2.2380`
